src/Query/Filter/Parameter.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query\Filter;
+
+use Doctrine\DBAL\ArrayParameterType;
+use Doctrine\DBAL\ParameterType;
+
+/** @internal */
+final class Parameter
+{
+    /** @param ParameterType::*|ArrayParameterType::*|string $type */
```

Comment:
With `ParameterType::*|ArrayParameterType::*`, your goal is to be more precise about the possible `int` we accept, when DBAL 3 is in use, right? But aren't you also removing `ParameterType` and `ArrayParameterType` from the possible types?


```suggestion
    /** @param ParameterType|ArrayParameterType|ParameterType::*|ArrayParameterType::*|string $type */
```

Reply:
No, I'm not. 😉

Reply:
Ah yes, `Type::*` will work for class constant and for enum cases as well, is that it?